client/plugins/fcc-source-challenges/create-challenge-nodes.js

```diff
@@ -29,7 +30,43 @@ function createChallengeNode(challenge, reporter) {
           internal,
           sourceInstanceName: 'challenge'
         },
-        challenge
+        challenge,
+        // if a field isEmpty, Gatsby does not include it on the node. However
+        // they need to exist (even if not used), this creates empty values for
+        // them.
+        {
+          required: isEmpty(challenge.required)
+            ? [{ link: '', src: '' }]
+            : challenge.required,
+          files: isEmpty(challenge.files)
+            ? {
```

Comment:
What about creating an object (i.e. `emptyFiles`) and stick it after the `?`.  Then before the return statement, you could use something like the following to create the object.
```
const emptyFiles = ['html', 'js', 'jsx'].reduce((obj, type) => {
  const properties = ['key', 'ext', 'name', 'contents', 'head', 'tail'];
  const keyValuesObj = properties.reduce((obj, prop) => ({ ...obj, [prop]: ''}), {});
  return { ...obj, [`index${type}`]: keyValuesObj };
}, {});
```

Reply:
Good point, it is rather ugly.  I'll sort that out tomorrow!

Reply:
I based my solution on yours, but decided against the nested reduce for reasons of readability.  Is this version okay?